src/lsm/segmented_array.zig

```diff
@@ -806,12 +806,31 @@ fn SegmentedArrayType(
                 assert(cursor.node == 0);
                 assert(cursor.relative_index == 0);
 
-                return Iterator{
+                return .{
                     .array = array,
                     .direction = direction,
                     .cursor = .{ .node = 0, .relative_index = 0 },
                     .done = true,
                 };
+            } else if (cursor.node == array.node_count - 1 and
+                cursor.relative_index == array.count(cursor.node))
+            {
+                return switch (direction) {
+                    .ascending => .{
+                        .array = array,
+                        .direction = direction,
+                        .cursor = .{ .node = 0, .relative_index = 0 },
```

Comment:
Is there any particular reason for resetting the cursor instead of preserving it + `done = true`?
```suggestion
                        .cursor = cursor,
```

Reply:
Nope, no difference. Updated.